model/riscv_sys_regs.sail

```diff
@@ -180,6 +180,22 @@ function haveUsrMode() -> bool = misa.U() == 0b1
 function haveNExt()    -> bool = misa.N() == 0b1
 /* see below for F and D extension tests */
 
+/* Cryptography extension support. Note these will need updating once */
```

Comment:
I hope this is a sensible way to put these functions in. The idea is that they will be updated when riscv-config / dynamic extension configuration is implemented in Sail. If not, please let me know where would be a better place.

I tried to replace the `sizeof(xlen) == X` with `cur_Architecture` calls, but it meant the `encdec` clauses had mismatched side-effects for some reason (`{reg}` v.s. `{reg, escape}`) and this is beyond my Sail knowledge to address.

Reply:
Yeah that comes from the `internal_error`. Not sure if that's quite the right way to implement that, might be better as an actual assertion as, as you note, it's a bit annoying. But this is fine, not your problem to solve, and everything else uses static XLEN checks.